### PR TITLE
Support rest parameters & default values

### LINKED ISSUE #7: Support rest parameters & default values
"fn-args": "^2.0.0",

Two bugs in one issue:

1. Rest parameter. Just try this:

const fnArgs = require("fn-args");

console.log(fnArgs((...arg1) => {})); // => []
console.log(fnArgs((arg1, ...arg2) => {})); // => [ 'arg1', '...arg2' ]

This is wrong as minimum because of ambiguity of rest parameter presence.

2. The reason I referred to Your package (my small internal argument getter works to this moment) is to skip default values:

console.log(fnArgs((arg1 = 2) => {})); // => []
console.log(fnArgs((arg1, arg2 = 2) => {})); // => [ 'arg1', 'arg2 = 2' ]
console.log(fnArgs((arg1 = 2, arg2) => {})); // => []

Of course, this is not what we expect, really?

Yeah, this is hell - to get rid of default values. But if You want to (I hope) remove this bug, try not to parse all 'default value code', but follow the number of open/close (, ), {, } and skip to the comma or parenthesis when no parenthesis depth occurs. Nevertheless, the code will be complicated dramatically.

### PR DESCRIPTION
**Pull request by GlaDos28**
Added support for rest parameter and default values.
Note that I leave some comments in the index.js. Moreover, there are two examples (as minimum) which I mentioned in commit - I think it is too unnecessary to fight against them.

The asymptotic time is **O(n)**, where n=function.toString().length, as it was.
Actual time of my **additional code** is probably about **n** (a bit less) because of one linear string scan. But it is only in theory, I don't want to waste time for analyzing the code - You can fix something yourself, if You want to :)

Some tests are attached.

*Hope it helps, then after merging, I use this package as a dependency :)*

---

Fixes #7

### CODE REVIEW

**Comment by GlaDos28** [comment]
Lastly remade code to pass by compiler. Note that I commented tests for default value because the compiler can't deal with default parameters, as I understood. Anyway, all tests have been passed.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+.idea
```
Comment (SamVerschueren) [inline .gitignore]: Remove this one

File: index.js [hunk 2]
```diff
@@ -1,20 +1,94 @@
 'use strict';
+
```
Comment (SamVerschueren) [inline index.js]: Remove empty line

**Comment by sindresorhus** [comment]
Sorry about the super slow response @GlaDos28. I just had too many PRs to deal with.

**Comment by sindresorhus** [comment]
This looks good. Thank you! :)

**Pull request merged**